encoding/protobuf.go

```diff
@@ -0,0 +1,98 @@
+package encoding
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/dynamic"
+	"github.com/yarpc/yab/transport"
+)
+
+// SplitMethod takes a method name like Service::Method and splits it
+// into Service and Method.
+func SplitMethod(fullMethod string) (svc, method string, err error) {
```

Comment:
Is there a common file for these things? Does this need to be public?

Reply:
Isn't used outside of this package so doesn't need to be public. From what I understand the `::` separator is a yarpc concept for splitting, not a thrift thing (although the thrift encoding package has a similar method). There is a general file, so if I'm correct this is a yarpc thing and not thrift I'll move this to the encoding file and change the proto and thrift implementations to point at that.

Reply:
Is `::` the right separator for protobuf methods? It's the scheme used by Thrift to join the Thrift service name with the method name. However, I think protobuf uses `/`.

I don't mind supporting `::` as well, but we should support the grpc native naming scheme too